components/nvs/include/nvs.hpp

```diff
@@ -70,6 +70,225 @@ const NvsErrCategory theNvsErrCategory{};
+  template <typename T>
+  void set(const char *key, T value, std::error_code &ec) {
+    check_lengths(ns_name, key, ec);
+    if (ec)
+      return;
+    esp_err_t err;
+    err = handle->set_item(key, value);
+    if (err != ESP_OK) {
+      ec = make_error_code(NvsErrc::Write_NVS_Failed);
+      logger_.error("Error {} writing to NVS!", esp_err_to_name(err));
+    }
+    return;
+  }
+
+  /// @brief Save a variable in the NVS
+  /// @param[in] key NVS Key of the variable to save
+  /// @param[in] value Variable to save
+  /// @param[out] ec Saves a std::error_code representing success or failure
+  template <typename T>
+  void set(std::string_view key, T value, std::error_code &ec) {
+    set(key.data(), value, ec);
+  }
+
+  /// @brief Set a bool in the NVS
+  /// @param[in] key NVS Key of the bool to set
+  /// @param[in] value bool to set
+  /// @param[out] ec Saves a std::error_code representing success or failure
+  void set(std::string_view key, bool value, std::error_code &ec) {
+    set(key.data(), value, ec);
+  }
+
+  /// @brief Set a bool in the NVS
+  /// @param[in] key NVS Key of the bool to set
+  /// @param[in] value bool to set
+  /// @param[out] ec Saves a std::error_code representing success or failure
+  void set(const char *key, bool value, std::error_code &ec) {
+    uint8_t u8 = static_cast<uint8_t>(value);
+    set<uint8_t>(key, u8, ec);
+  }
+
+  /// @brief Set a string in the NVS
+  /// @param[in] key NVS Key of the string to set
+  /// @param[in] value string to set
+  /// @param[out] ec Saves a std::error_code representing success or failure
+  void set(std::string_view key, const std::string &value,
+               std::error_code &ec) {
+    set(key.data(), value, ec);
+  }
+
+  /// @brief Set a string in the NVS
+  /// @param[in] key NVS Key of the string to set
+  /// @param[in] value string to set
+  /// @param[out] ec Saves a std::error_code representing success or failure
+  void set(const char *key, const std::string &value,
+               std::error_code &ec) {
+    check_lengths(ns_name, key, ec);
+    if (ec)
+      return;
+    esp_err_t err;
+    err = handle->set_string(key, value.data());
+    if (err != ESP_OK) {
+      ec = make_error_code(NvsErrc::Write_NVS_Failed);
+      logger_.error("Error {} writing to NVS!", esp_err_to_name(err));
+      return;
+    }
+    return;
+  }
+
+  /// @brief Commit changes
+  /// @param[out] ec Saves a std::error_code representing success or failure
+  /// @details Commits changes to the NVS
+  void commit(std::error_code &ec) {
+    esp_err_t err = handle->commit();
+    if (err != ESP_OK) {
+      logger_.error("Error {} committing to NVS!", esp_err_to_name(err));
+      ec = make_error_code(NvsErrc::Commit_NVS_Failed);
+    }
+    return;
+  }
+
+protected:
+  std::unique_ptr<nvs::NVSHandle> handle;
+  const char *ns_name;
+
+  void check_lengths(const char *ns_name, const char *key, std::error_code &ec) {
```

Comment:
you should update this function to not take in / check `ns_name`, since that variable doesn't change after construction, but this function is called every time an operation is performed on the handle.